project-koku/nise
# update to oci generator

# changes
1. add tags to yaml generator
2. allow environment variables fallback if no config file

# testing instructions

1. to test tags in yaml file;
   - run `nise yaml oci -o test-oci.yml`
   - you should see tags key in all generators, see example below
```
    
---
generators:
  - OCIComputeGenerator:
      start_date: 2022-08-01
      end_date: 2022-09-26
      cost: 0.31624
      currency: USD
      compartment_name: ethanbowers
      tenant_id: ocid1.tenancy.oc1..EfjkUPxyZSYLvd
      tags:
        tags/free-form-tag: score
        tags/orcl-cloud.free-tier-retained: resource

  - OCIBlockStorageGenerator:
      start_date: 2022-08-01
      end_date: 2022-09-26
      cost: 0.40168
      currency: USD
      compartment_name: ethanbowers
      tenant_id: ocid1.tenancy.oc1..EfjkUPxyZSYLvd
      tags:
        tags/new-tags.tarnished-tags: street
        tags/orcl-cloud.free-tier-retained: really

  - OCINetworkGenerator:
      start_date: 2022-08-01
      end_date: 2022-09-26
      cost: 0.3966
      currency: USD
      compartment_name: ethanbowers
      tenant_id: ocid1.tenancy.oc1..EfjkUPxyZSYLvd
      tags:
        tags/free-form-tag: down

  - OCIDatabaseGenerator:
      start_date: 2022-08-01
      end_date: 2022-09-26
      cost: 0.4873
      currency: USD
      compartment_name: ethanbowers
      tenant_id: ocid1.tenancy.oc1..EfjkUPxyZSYLvd
      tags:
        tags/free-form-tag: hear

```
